be/src/olap/rowset/rowset.h

```diff
@@ -95,6 +151,37 @@ class Rowset : public std::enable_shared_from_this<Rowset> {
     // TODO should we rename the method to remove_files() to be more specific?
     virtual OLAPStatus remove() = 0;
 
+    // close to clear the resource owned by rowset
+    // including: open files, indexes and so on
+    // NOTICE: can not call this function in multithreads
+    void close() {
+        RowsetState old_state = _rowset_state_machine.rowset_state();
+        if (old_state == ROWSET_UNLOADED) {
```

Comment:
When the state is unloading, seems OK to return

Reply:
yes, u are right.